src/query/query_parser/query_parser.rs

```diff
@@ -591,6 +616,19 @@ mod test {
         assert_eq!(format!("{:?}", query), "EmptyQuery");
     }
 
+    #[test]
+    pub fn test_parse_query_empty_match_all() {
+        test_parse_query_to_logical_ast_helper("", "<emptyclause>", false);
+        test_parse_query_to_logical_ast_helper(" ", "<emptyclause>", false);
+        let mut query_parser = make_query_parser();
```

Comment:
Why wasn't it possible to work off the logical ast to test the match all case too?

Reply:
Lack of confidence with the language mostly. I tried first with a "good enough" solution to see if i could do it. Now I'll try to improve on it. Thanks for the patience

Reply:
This has not been fixed.

Reply:
sorry I forgot! :(